app/src/main/java/com/concordia/mcga/utilities/pathfinding/IndoorDirections.java

```diff
@@ -0,0 +1,305 @@
+package com.concordia.mcga.utilities.pathfinding;
+
+import com.concordia.mcga.models.IndoorMapTile;
+
+import java.util.List;
+
+
+public class IndoorDirections {
+
+    private static final int UNDEF_ORIENTATION = 0;
+    private static final int NORTH_ORIENTATION = 1;
+    private static final int EAST_ORIENTATION = 2;
+    private static final int SOUTH_ORIENTATION = 3;
+    private static final int WEST_ORIENTATION = 4;
+
+    private int orientation;
+
+    ///////////////
+    // Main Code //
+    ///////////////
+
+    /**
+     * Constructor
+     */
+    public IndoorDirections(){
+        orientation = UNDEF_ORIENTATION;
+    }
+
+    /**
+     * Get list of directions for the user
+     * @param tiles
+     * @return String to tell user what to do as well as an image
+     */
+    public String[][] getDirections(List<IndoorMapTile> tiles){
+        String currentDirection = null;
+        int distance = 0;
+        String turn = null;
+        String image = null;
+
+        // Size -1 because we don't need directions to the first point
+        String[][] returnString = new String[tiles.size() - 1][2];
+
+        int x1, x2, y1, y2;
+        try {
+            // set initial coordinates once
+            int previousCoordinateX = tiles.get(0).getCoordinateX();
+            int previousCoordinateY = tiles.get(0).getCoordinateY();
+
+            // Inittial orientation
+            initOrientation(tiles.get(0).getCoordinateX(), tiles.get(1).getCoordinateX(), tiles.get(0).getCoordinateY(), tiles.get(1).getCoordinateY());
+
+            for (int i = 1; i < tiles.size(); i++) {
+                x1 = previousCoordinateX;
+                x2 = tiles.get(i).getCoordinateX();
+                y1 = previousCoordinateY;
+                y2 = tiles.get(i).getCoordinateY();
+
+                turn = getTurn(x1, x2, y1, y2);
+                distance = MeasureDistance(x1, x2, y1, y2);
+                image = getImage();
+
+                // Format looks like: Turn Right In 200U ; where 'U' is for Unit
+                currentDirection = "Turn " + turn + " In " + distance + "u";
```

Comment:
Ideally we would have an objective measurement here. Is it possible to approximate distance in m for a floor plan?